### ISSUE TITLE
Feature Request: Add ability to have multiple .repo folders

### ISSUE THREAD

**Issue by parthea** [googleapis/github-repo-automation#493]
I'm a new user of the `repo` cli. In my workflow, I found it could be helpful to have multiple `.repo` folders when multitasking on 2 unrelated tasks that are still work in progress instead of just the [one](https://github.com/googleapis/github-repo-automation/blob/master/src/sync.ts#L49) `~/.repo`. 

The `cli` already looks for a `config.yaml` file in the current working directory. Running `repo sync` could clone repositories in the current directory as well so that `git reset --hard` is less likely to erase changes. In this setup, I would maintain multiple copies of repositories. 

As a workaround, I could use branches to maintain my work in progress, but having multiple `config.yaml` files would allow me have a better multitasking workflow.

If the recommended solution is to use branches to maintain work in progress, please feel free to close this issue. Perhaps I just need to get used to the fact that `repo sync` is analogous to `git reset --hard` and `git pull`.

In the current design:
- Multiple config.yaml files are supported
- Running `repo sync` will only sync repositories that match criteria in `config.yaml` file in the current working directory
- `repo exec -- git status` will result in running `git status` on all repositories in `~/.repo`

The proposed design:
- Multiple config.yaml files are supported
- Running `repo sync` will only sync repositories that match criteria in `config.yaml` file in the current working directory
- `repo exec -- git status` will result in running `git status` on all repositories in `<current working directory>/.repo`

In other words, `repo exec` will only run on repositories that you are actively working on.

**Comment by parthea** [comment]
I found a workaround. For multitasking work in progress tasks, I'm planning to temporarily move my local `.repo` folder to another path other than `~/` and let the `cli` create a new .repo folder in '~/.repo`.
